packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -347,6 +348,36 @@ Future<void> buildGradleProject({
   }
 }
 
+// TODO(mklim): Fix the obvious bugs.
+// - Do we need logic for passing along build architecture, local engine, etc?
+// - Make this smart enough to not rebuild the plugin every time?
```

Comment:
Shouldn't gradle take care of that?

I'd time how long that takes, and if it's more than a few ms then we should just be able to check for the build output ourselves.